docs/design/proposals/codegen-refactor/model_cmd_2.md

```diff
@@ -0,0 +1,143 @@
+* Overloaded functions such as [SetResource](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/generate/code/set_resource.go#L74); responsible for a lot more than generating Go code
+* Confusing CX. There's a not-so-clear dependency on a folder in the $SERVICE-controller via [getLatestAPIVersion](https://github.com/aws-controllers-k8s/code-generator/blob/26e5da2e7656bb836ee438c05df14f2adc50197d/cmd/ack-generate/command/common.go#L271). This goes against the flow of the `pipeline` and creates confusion as there is a `--version` flag for the `apis` command (the above function takes precedence).
+* Long dev loop; after making changes to either *generator.yaml* or logic in *code-generator* users need to either manually verify files or run e2e tests-- both of which are time consuming today
+* How *code-generator* reconciles `ackgenconfig` with `aws-sdk` to create `ackmodel` is inconsistent (and some duplicated work). For example, overrides such as `custom_shapes` [edit aws-sdk directly](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/sdk/custom_shapes.go#L62) while others such as `ignored_operations` [set operation values to nil](https://github.com/aws-controllers-k8s/code-generator/blob/26e5da2e7656bb836ee438c05df14f2adc50197d/pkg/model/model.go#L295) in memory leaving the actual operations in `aws-sdk` untouched.
+
+Current Pipeline
+---
+
+![current-pipeline](./images/current_pipeline.png)
+* Displayed are the important and mostly duplicated calls between the `apis` and `controller` commands
+* The cache is only used to store the `aws-sdk` repo
+* The red line denotes a dependency on $SERVICE-controller (output directory) repo that should be removed
+* Only `GetCRDs` is duplicated between the commands when creating the model
+
+
+Current Dev Loop
+---
+
+![current-dev-loop](./images/current_dev_loop.png)
+* Validating the result of a change in *code-generator* and/or *generator.yaml* can be a lengthy process
+* In the current feedback loop, users and contributors need to manually verify files or run lengthy e2e tests even for minor changes to `ackgenconfig`
+
+
+## Solution
+Add a new command, `model`, which takes in `aws-sdk` and *generator.yaml* as inputs to create and cache `ackmodel`. Existing commands, `api` and `controller`, will remove parameters for `aws-sdk` and *generator.yaml* and replace them with `ackmodel` input and/or let it default to `--cache-dir`.
+
+Updated Pipeline
+---
+![updated-pipeline](./images/proposed_pipeline.png)
+* This is not an exhaustive diagram of the calls, but shows the clear responsibility of `generateModel` and how downstream commands like `api` and `controller` become significantly lighter and easier to follow.
+* `ackmodel` will be cached in the same folder as `aws-sdk` and downstream commands will check for `ackmodel` instead of `aws-sdk`
+* users can override with a path to their own `ackmodel` (for even quicker testing!), but by default will check `--cach-dir`
```

Comment:
When would users have their own `ackmodel`? Would users not already have their own `generator.yaml`?

Reply:
This would mainly be for power users/experimentation, but in general users wouldn't come with their own `ackmodel` but generate a valid `ackmodel` as a base, then work on top of that. There wouldn't be many scenarios, but one example is adding a new config to *generator.yaml*. One could directly edit a CRD's Spec/Status field or a Shape in `typeDefs` assuming the new config "worked", then generate apis/controller to ensure a valid/working approach.